File: csharp/rotational-cipher/RotationalCipher.cs
using System;

public static class RotationalCipher
{
    public static string Rotate(string text, int shiftKey)
    {
        char[] tmpStr = text.ToCharArray();
        for (int i=0; i<tmpStr.Length; i++)
        {
            tmpStr[i] = RotateChar(tmpStr[i], shiftKey);
        }
        return new string(tmpStr);
    }

    private static char RotateChar(char c, int key)
    {
        if (!char.IsLetter(c))
            return c;
        int amtToShift;
        bool isUpper = Char.IsUpper(c);
        amtToShift = !isUpper ? c - 'a' + key : c - 'A' + key;
        amtToShift = amtToShift > 25 ? amtToShift - 26 : amtToShift;
        return (char)(isUpper ? 'A' + amtToShift : 'a' + amtToShift);
    }
}

File: java/rotational-cipher/src/main/java/RotationalCipher.java
import java.util.stream.*;
class RotationalCipher {

    private int shiftKey;
    RotationalCipher(int shiftKey) {
        this.shiftKey = shiftKey;
    }

    String rotate(String data) {
        return data.chars().map(c -> {
            if (!Character.isAlphabetic(c)) return c;
            int tmpChar = Character.isLowerCase(c) ? c-'a' : c-'A';
            tmpChar = (tmpChar+shiftKey)%26;
            tmpChar = Character.isLowerCase(c) ? tmpChar + 'a' : tmpChar + 'A';
            return (char)tmpChar;
        }).collect(StringBuilder::new,
                StringBuilder::appendCodePoint,
                StringBuilder::append)
                .toString();
    }
}
